### PR TITLE
Spill stats

### PR DESCRIPTION
**Pull request by pnowojski**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/spiller/SpillerFactoryWithStats.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spiller;
+
+import org.weakref.jmx.Managed;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public abstract class SpillerFactoryWithStats
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/spiller/SpillerFactoryWithStats.java:20]: does it make sense to distinguish `SpillerFactory` and `SpillerFactoryWithStats`? I think stats should always be available.

File: presto-main/src/main/java/com/facebook/presto/spiller/SpillerFactory.java [hunk 2]
```diff
@@ -15,10 +15,14 @@
...
 
 import com.facebook.presto.spi.type.Type;
+import org.weakref.jmx.Managed;
 
 import java.util.List;
 
 public interface SpillerFactory
 {
+    @Managed
+    long getSpilledData();
```
Comment (sopel39) [inline presto-main/src/main/java/com/facebook/presto/spiller/SpillerFactory.java:25]: `getSpilledDataSize()`?

File: presto-main/src/main/java/com/facebook/presto/spiller/SpillerFactoryWithStats.java (hunk 1 above)
Comment (pnowojski) [reply presto-main/src/main/java/com/facebook/presto/spiller/SpillerFactoryWithStats.java:20]: Stats are always visible. `getSpilledData` is defined in `SpillerFactory` interface. This is just an abstract implementation of Factory that provides stats.

File: presto-main/src/main/java/com/facebook/presto/spiller/SpillerFactory.java (hunk 2 above)
Comment (pnowojski) [reply presto-main/src/main/java/com/facebook/presto/spiller/SpillerFactory.java:25]: changed

**Pull request closed without merging**